x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/rest/RestSamlGenerateMetadataRequest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.idp.rest;
+
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.action.RestBuilderListener;
+import org.elasticsearch.xpack.idp.action.SamlGenerateMetadataAction;
+import org.elasticsearch.xpack.idp.action.SamlGenerateMetadataRequest;
+import org.elasticsearch.xpack.idp.action.SamlGenerateMetadataResponse;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.GET;
+
+public class RestSamlGenerateMetadataRequest extends BaseRestHandler {
+
+    static final ObjectParser<SamlGenerateMetadataRequest, Void> PARSER =
+        new ObjectParser<>("idp_generate_metadata", SamlGenerateMetadataRequest::new);
+
+    static {
+        PARSER.declareString(SamlGenerateMetadataRequest::setSpEntityId, new ParseField("sp_entity_id"));
```

Comment:
Did you have a reason for putting this in the body rather than as a URL/query parameter?

Reply:
No, I think I had this as a POST originally so that I wouldn't have to worry about the ugliness of 
```
GET https://node1.example.com/_idp/saml/sp/https%3A%2F%2Fsp.entityid.org
```

On the other hand, GET makes sense.  I assume we will be doing the same in the registration API ( with PUT in that case ). Thoughts?

Reply:
GET it is.

Reply:
GET it is.

Reply:
What you've got in the most recent commit is what I expected.